exercises/practice/dnd-character/dnd_character_test.rb

```diff
@@ -0,0 +1,102 @@
+require 'minitest/autorun'
+require_relative 'dnd_character'
+
+module Helpers
+  BASE_HITPOINTS = 10
+
+  def attributes
+    %i[strength dexterity constitution intelligence wisdom charisma]
+  end
+end
+
+class DndCharacterTest < Minitest::Test
+  include Helpers
+
+  def test_modifier_score_3
+    assert_equal DndCharacter.modifier(3), -4
+  end
+
+  def test_modifier_score_4
+    assert_equal DndCharacter.modifier(4), -3
+  end
+
+  def test_modifier_score_5
+    assert_equal DndCharacter.modifier(5), -3
+  end
+
+  def test_modifier_score_6
+    assert_equal DndCharacter.modifier(6), -2
+  end
+
+  def test_modifier_score_7
+    assert_equal DndCharacter.modifier(7), -2
+  end
+
+  def test_modifier_score_8
+    assert_equal DndCharacter.modifier(8), -1
+  end
+
+  def test_modifier_score_9
+    assert_equal DndCharacter.modifier(9), -1
+  end
+
+  def test_modifier_score_10
+    assert_equal DndCharacter.modifier(10), 0
+  end
+
+  def test_modifier_score_11
+    assert_equal DndCharacter.modifier(11), 0
+  end
+
+  def test_modifier_score_12
+    assert_equal DndCharacter.modifier(12), 1
+  end
+
+  def test_modifier_score_13
+    assert_equal DndCharacter.modifier(13), 1
+  end
+
+  def test_modifier_score_14
+    assert_equal DndCharacter.modifier(14), 2
+  end
+
+  def test_modifier_score_15
+    assert_equal DndCharacter.modifier(15), 2
+  end
+
+  def test_modifier_score_16
+    assert_equal DndCharacter.modifier(16), 3
+  end
+
+  def test_modifier_score_17
+    assert_equal DndCharacter.modifier(17), 3
+  end
+
+  def test_modifier_score_18
+    assert_equal DndCharacter.modifier(18), 4
+  end
+
+  def test_random_character_stats
+    100.times do
+      character = DndCharacter.new
+      allowed_range = (3..18)
+      expected_hitpoints = BASE_HITPOINTS + DndCharacter.modifier(character.constitution)
+
+      attributes.each do |attribute|
+        assert_includes allowed_range, character.send(attribute)
+      end
+
+      assert_equal expected_hitpoints, character.hitpoints
```

Comment:
This one is in the correct order, the expected first, and then character hitpoints being actual.

Indeed, we could remove the `expected_` from the name, or we could say only `expected` because the actual shows what kind of thing is expected based on the method being called on the object under test.